src/Modules/ModuleTypes.jl

```diff
@@ -4,8 +4,8 @@ import AbstractAlgebra.WeakKeyIdDict
 # we expect the `ModuleFP` framework to be functional and/or working.
 # It can gradually be extended, but should not be considered to be 
 # visible or accessible to the outside world. 
-const AdmissibleModuleFPRingElem = Union{RingElem, PBWAlgQuoElem, PBWAlgElem}
-const AdmissibleModuleFPRing = Union{Ring, PBWAlgQuo, PBWAlgRing}
+const AdmissibleModuleFPRingElem = NCRingElem
+const AdmissibleModuleFPRing = NCRing
```

Comment:
I was looking into how to **not** widen the signatures of `AdmissibleRingElem`s for the `ModuleFP`s to all of `NCRingElem`, but failed for the following reasons. So far, we declare the `AdmissibleRingElem`s as `const`. That means we have to include all types we would like to put into this union up front. In the case of `ExtAlgElem` and its parent, this is bad, because they use `FreeMod`s in their internals! So the cat bites its tail here and I haven't yet figured out a good way to resolve this. 

Either way, we need to resolve this before merging!